src/common/utils.sh

```diff
@@ -13,9 +13,9 @@ pid_guard() {
   name=$2
 
   if [ -f "$pidfile" ]; then
-    pid=$(head -1 "$pidfile")
-    echo "pidno" $pid
-    if [ -n "$pid" ] && [ -e /proc/$pid ]; then
+    pid="$(head -1 $pidfile)"
+    echo "pidno $pid"
+    if [ -n "$pid" ] && [ -e "/proc/$pid" ] && [[ "$(readlink /proc/$pid/exe)" =~ ^(.*/)?$name$ ]]; then
```

Comment:
This does not work for the `scheduler` process which is run by `java`:

```
/var/vcap/jobs/scheduler# readlink /proc/$pid/exe
/var/vcap/data/packages/openjdk-11/80438e54d5b5da48474fb42ae0427e4454128fb8/bin/java
```
Maybe you could fall back to to checking `/proc/$pid/cmdline` for a match?

```
/var/vcap/jobs/scheduler# strings /proc/$pid/cmdline 
java
-Dspring.config.location=/var/vcap/jobs/scheduler/config/application.properties
-jar
/var/vcap/packages/scheduler/scheduler-1.0-SNAPSHOT.war
```

Reply:
You are right.... I changed the code.
Thank you